### PR TITLE
Resolve cv-lint issues

### PR DESCRIPTION
**Pull request by swati-nair**
1. Added app.kubernetes.io labels on all the resources in deploy directory.
2. Added nodeAffinity in operator deployment
3. Added CPU and memory resource limits and requests in operator container
4. Added securityContext for operator container

### CODE REVIEW

File: deploy/operator.yaml [hunk 1]
```diff
@@ -11,13 +15,27 @@ spec:
...
         productVersion: "3.3.0"
         productMetric: FREE
     spec:
       serviceAccountName: ibm-cert-manager-operator
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: beta.kubernetes.io/arch
```
Comment (tthorpe2) [inline deploy/operator.yaml:33]: key: beta.kubernetes.io/arch shows as deprecated in kube documentation - states one should use - key: beta.kubernetes.io/arch

File: deploy/operator.yaml [hunk 2]
```diff
@@ -37,3 +55,17 @@ spec:
                   fieldPath: metadata.name
             - name: OPERATOR_NAME
               value: "ibm-cert-manager-operator"
+          resources:
```
Comment (tthorpe2) [inline deploy/operator.yaml:58]: how did you derive the limits specified?

File: deploy/operator.yaml [hunk 3]
```diff
@@ -37,3 +55,17 @@ spec:
...
+            limits:
+              cpu: 20m
+              memory: 90Mi
+            requests:
+              cpu: 10m
+              memory: 50Mi
+          securityContext:
```
Comment (tthorpe2) [inline deploy/operator.yaml:65]: Did the ibm cert guidelines specify these settings? Any mention of the contrast beween this and running on the ocp default restricted scc? The restricted scc for openshift is similar but does not seem to match exactly.  And did you find doc on capabilities: drop: - ALL    I can't see to find a syntax reference for this.

File: deploy/operator.yaml (hunk 2 above)
Comment (swati-nair) [reply deploy/operator.yaml:58]: I came up with these values by checking how much does cert-manager pod usually consumes these resources in grafana (after installing monitoring)

File: deploy/operator.yaml (hunk 3 above)
Comment (swati-nair) [reply deploy/operator.yaml:65]: So, I got this error while running cv-lint: 
```deploy/operator.yaml: "ALL" not found in spec.template.spec.containers[0].securityContext.capabilities.drop```

And
```
deploy/operator.yaml: keys for pod security context not defined: spec.template.spec.hostNetwork, spec.template.spec.hostPID, spec.template.spec.hostIPC, spec.template.spec.securityContext.runAsNonRoot OR spec.template.spec.containers[0].securityContext.runAsNonRoot, spec.template.spec.securityContext.runAsUser OR spec.template.spec.containers[0].securityContext.runAsUser, spec.template.spec.containers[0].securityContext.privileged, spec.template.spec.containers[0].securityContext.readOnlyRootFilesystem, spec.template.spec.containers[0].securityContext.allowPrivilegeEscalation, at least one element in spec.template.spec.containers[0].securityContext.capabilities.add OR spec.template.spec.containers[0].securityContext.capabilities.drop
```

I deployed and tested the operator with the new parameters on my cluster successfully without any issue, and saw that the default scc is still `openshift.io/scc: restricted`. 
For `drop: - ALL` , I compared it with cert-manager helm chart params  where we have set the security params (https://github.ibm.com/IBMPrivateCloud/icp-cert-manager-chart/blob/24e3994eb42abda660645be75f8797b19191bc67/stable/ibm-cert-manager/templates/deployment.yaml#L142) and also verified it with metering-operator changes

File: deploy/operator.yaml (hunk 1 above)
Comment (swati-nair) [reply deploy/operator.yaml:33]: Discussed with Steve, his assumption was that using `kubernetes.io/arch` can create problems with OCP3.11, so he has added this in metering-operator. But, I can also add one more condition to check for `kubernetes.io/arch`
The cv-lint error was: `deploy/operator.yaml: one of ["beta.kubernetes.io/arch" "kubernetes.io/arch"] at some spec.template.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[i].matchExpressions[j].key not defined for architecture-based node affinity`

Comment (tthorpe2) [reply deploy/operator.yaml:33]: ok..that explanation makes sense.

File: deploy/operator.yaml (hunk 2 above)
Comment (tthorpe2) [reply deploy/operator.yaml:58]: You mean the operator pod?

Comment (swati-nair) [reply deploy/operator.yaml:58]: yes

**Pull request merged**